client/src/components/HeaderComponent/index.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import logo from '../../assets/images/logo.svg';
+import Button from '../Button/index';
+import './style.css';
+
+export default class Header extends React.Component {
+  state = {
+    menuIsShown: false,
+  };
+
+  onClickHandler = () => {
```

Comment:
just a minor thing here, I prefer specific descriptive names - instead of `onClickHandler()` it could be called `toggleMenuIsShown()` or `toggleMenu()`. Then below in the button it would look like
```js
<Button className="burger" onClick={this.toggleMenu}>
```
It's much more obvious what happens when you read the code, don't have to navigate to the onClickHandler method to find out...